### PR TITLE
Draft release notes for 0.6.0

### PR DESCRIPTION
**Pull request by lucafavatella**
Also:

* Update version numbers in doc;

* Avoids usage of expression "this location" - addressing https://github.com/aeternity/epoch/pull/591#issuecomment-358350411

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -65,7 +63,7 @@ Create a directory and unpack the downloaded package:
 ```
 mkdir /tmp/node
 cd /tmp/node
-tar xf ~/Downloads/epoch-0.5.0-osx-10.12.6.tar.gz
+tar xf ~/Downloads/epoch-0.6.0-osx-10.12.6.tar.gz
```
Comment (lucafavatella) [inline RELEASE-NOTES.md]: @dincho Is this suffix 10.12.6 confirmed? (Not a big deal - just for not confusing users.)

Comment (dincho) [reply RELEASE-NOTES.md]: yes, you can take a look https://circleci.com/gh/aeternity/epoch/442#artifacts/containers/0
However, we'll upload the binaries manually (or by still running Travis, if it gets it's job on time)

**Comment by dincho** [comment]
Scary PR number 😈

**Comment by lucafavatella** [comment]
CI for last PR merge completed. I see two approvals.

**Pull request merged**